usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -1297,6 +1300,141 @@ CK_RV ep11tok_pkey_add_protkey_attr_to_tmpl(TEMPLATE *tmpl)
 done:
     return ret;
 }
+
+static CK_RV ep11tok_pkey_convert_key(STDLL_TokData_t *tokdata, SESSION *session,
+                                     OBJECT *key_obj, CK_BBOOL xts_mode,
+                                     CK_BYTE *protkey, CK_ULONG *protkey_len)
+{
+    ep11_private_data_t *ep11_data = tokdata->private_data;
+    int num_retries = 0, vp_offset1, vp_offset2;
+    CK_ATTRIBUTE *skey_reenc_attr = NULL;
+    CK_ATTRIBUTE *skey_attr = NULL;
+    CK_ATTRIBUTE *pkey_attr = NULL;
+    CK_RV rc;
+
+    if (*protkey_len < MAXPROTKEYSIZE - 16) {
+        rc = CKR_BUFFER_TOO_SMALL;
+        goto done;
+    }
+
+    /* Try to obtain a write lock on the key_obj */
+    rc = object_unlock(key_obj);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("object_unlock failed, rc=0x%lx\n", rc);
+        goto done;
+    }
+
+    rc = object_lock(key_obj, WRITE_LOCK);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Could not obtain write lock.\n");
+        goto done;
+    }
+
+    /*
+     * The key_obj could be modified between unlock and lock. Therefore get
+     * the attribute when we have the write lock here.
+     */
+    if (template_attribute_get_non_empty(key_obj->template,
+                                         CKA_IBM_OPAQUE, &skey_attr) != CKR_OK) {
+        TRACE_ERROR("This key has no blob: should not occur!\n");
+        rc = CKR_FUNCTION_FAILED;
+        goto done;
+    }
+
+    /* Convert secure key to protected key. */
```

Comment:
Move that comment to below the retry label, it seems to fit better there.

Reply:
fixed.